pkg/tsdb/prometheus/prometheus.go

```diff
@@ -220,7 +218,12 @@ func parseResponse(value model.Value, query *PrometheusQuery) (*tsdb.QueryResult
 	return queryRes, nil
 }
 
-func wrapAsPrometheusError(err error) PrometheusError {
+func IsAPIError(err error) bool {
+	var e *apiv1.Error
+	return errors.As(err, &e)
```

Comment:
`errors.Is` is the function that should be used to tell if an error is of certain type (or wraps it).
```suggestion
	return errors.Is(err, &apiv1.Error{})
```